pkg/prometheus/watcher.go

```diff
@@ -660,32 +749,67 @@ func (w *Watcher) syncAlertManagerConfig(key string) error {
 		return nil
 	}
 
-	log.Debugf("syncing alert manager config: key: %s, name: %s", key, amc.Spec.Type)
-
+	log.Debugf("syncing alert manager config: key: %s, type: %s", key, amc.Spec.Type)
 	for key, val := range amc.Labels {
 		log.Debugf("Labels: %s %s", key, val)
 		if key == "alertmanager" {
 			alertManagerName = val
 			break
 		}
 	}
+	log.Debugf("Alert manager label: %s", alertManagerName)
 
 	if alertManagerName == "" {
 		log.Errorf("Alert manager label missing in alertmanager config: %s", key)
 		return nil
 	}
 
-	switch amc.Spec.Type {
-	case "slack":
-		log.Debug("Formatting slack configuration")
-		data, err = w.formatAlertConfig(amc.Spec.Params)
-		if err != nil {
-			log.Errorf("Failed to format alert config for: %s", key)
-			return nil
+	file, err := os.Open(configDir + "/alertmanager.yaml")
+	if err != nil {
+		log.Error("Failed to open alert manager config file", err)
+		return os.ErrInvalid
+	}
+	defer file.Close()
+	data, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Error("Failed to read alert manager config file", err)
+		return os.ErrInvalid
+	}
+
+	var acfg alertConfig
+	yaml.Unmarshal(data, &acfg)
+
+	err = w.formatReceiver(amc, &acfg)
+	if err != nil {
+		log.Errorf("Failed to format receiver for: %s", amc.Spec.Type)
+		return err
+	}
+
+	var options metav1.ListOptions
+	var amcList *monitoringv1.AlertmanagerConfigList
+	amcList, err = w.mClient.MonitoringV1().AlertmanagerConfigs(metav1.NamespaceAll).List(options)
```

Comment:
Alertmanager is namespaced right? Why read alertmanager configs from all namespaces?

Reply:
Yes, right now all objects will be deployed in pf9-monitoring ns, so we can just watch that. 

But the shared index informers loops watch all namespaces, added all namespaces here, is there a possibility that we might want to deploy in any other namespaces. If not will fix all such occurrences to pf9-monitoring.

Reply:
Also, since customer created alert managers (from the UI) can be in any namespace, we need to watch all namespaces to create a secret accordingly.